src/section/section_authority_provider.rs

```diff
@@ -61,6 +65,18 @@ impl SectionAuthorityProvider {
     pub(crate) fn position(&self, name: &XorName) -> Option<usize> {
         self.elders.keys().position(|other_name| other_name == name)
     }
+
+    pub(crate) fn set_pk_set(&mut self, pk_set: ReplicaPublicKeySet) {
+        self.pk_set = Some(pk_set);
+    }
+
+    pub(crate) fn clone_without_pk_set(&self) -> Self {
```

Comment:
Without the Section key this is useless as an Elder with no Section Key to identify him is unable to tell us anything. We should never have this and if we use it that place has bugs ;-)

Reply:
The place we use this is at https://github.com/maidsafe/sn_routing/blob/master/src/routing/core/messaging/handling/agreement.rs#L192
when handling SectionInfo, to ensure the `agreement` is not about an out-dated candidates.
Stuff from `promote_and_demote_elders` cannot to have key-related info.
Hence this mapping util function is required.

Reply:
However Adults,clients and Elders will now all hold the SAP for the current key. So can we now use that? So promote / demote etc. have this and get a new one and can see if they are Edlers or not from the difference?